src/main/java/svc/data/textMessages/SMSAlertDAO.java

```diff
@@ -0,0 +1,164 @@
+		Map<String, Object> parameterMap = new HashMap<String, Object>();
+		parameterMap.put("citationNumber", citationNumber);
+		parameterMap.put("phoneNumber", phoneNumber);
+		String sql = "SELECT * FROM sms_alerts WHERE citation_number = :citationNumber AND phone_number = :phoneNumber";
+		try{
+			jdbcTemplate.queryForObject(sql, parameterMap, new SMSAlertSQLMapper());
+			return true;
+		}catch(IncorrectResultSizeDataAccessException irsdae){
+			return false;
+		}catch(Exception e){
+			LogSystem.LogDBException(e);
+			return false;
+		}
+	}
+	
+	public boolean add(String citationNumber, LocalDateTime courtDateTime, String phoneNumber, LocalDate dob){
+		if (!doesAlertExist(citationNumber,phoneNumber)){
+			Map<String, Object> parameterMap = new HashMap<String, Object>();
+			parameterMap.put("citationNumber", citationNumber);
+			parameterMap.put("courtDateTime", DatabaseUtilities.convertLocalDateTimeToDatabaseDate(courtDateTime));
+			parameterMap.put("phoneNumber", phoneNumber);
+			parameterMap.put("dob", DatabaseUtilities.convertLocalDateToDatabaseDate(dob));
+			String sql = "INSERT INTO sms_alerts (citation_number,court_date,phone_number,date_of_birth) VALUES (:citationNumber,:courtDateTime,:phoneNumber,:dob)";
+			try{
+				int rowsAffected = jdbcTemplate.update(sql, parameterMap); 
+				if (rowsAffected==1){
+					return true;
+				}else{
+					return false;
+				}
+			}catch(Exception e){
+				LogSystem.LogDBException(e);
+				return false;
+			}	
+		}else{
+			return true; //entry already exists, so return as if it were newly inserted
+		}
+	}
+	
+	public boolean remove(String citationNumber, String phoneNumber, LocalDate dob){
+		if (doesAlertExist(citationNumber,phoneNumber)){
+			Map<String, Object> parameterMap = new HashMap<String, Object>();
+			parameterMap.put("citationNumber", citationNumber);
+			parameterMap.put("phoneNumber", phoneNumber);
+			parameterMap.put("dob", DatabaseUtilities.convertLocalDateToDatabaseDate(dob));
+			String sql = "DELETE FROM sms_alerts WHERE citation_number=:citationNumber  AND phone_number=:phoneNumber AND date_of_birth=:dob";
+			try{
+				jdbcTemplate.execute(sql, parameterMap, new PreparedStatementCallback<Boolean>(){
+
+					@Override
+					public Boolean doInPreparedStatement(PreparedStatement ps)
+							throws SQLException, DataAccessException {
+						return ps.execute();
+					}
+					
+				});
+				return true;
+			}catch(Exception e){
+				LogSystem.LogDBException(e);
+				return false;
+			}
+		}else{
+			return true;  //entry did not exist, to return as if it were removed
+		}
+	}
+	
+	public void removeExpiredAlerts(){
+		LocalDateTime expiredDate = DatabaseUtilities.getCurrentDateTime();
+		String sql = "DELETE FROM sms_alerts WHERE court_date < '"+DatabaseUtilities.convertLocalDateTimeToDatabaseDateString(expiredDate.plusDays(2))+"'";
+		
+		jdbcTemplate.execute(sql, new PreparedStatementCallback<Boolean>(){
```

Comment:
What's this prepared statement business? This looks a little verbose compared to what I've seen in the past for deletes in jdbc